File: JavaScript/IsAnagram.js
function Isanagram(s1, s2) {
    let freq = {}
    if (s1.length !== s2.length) return false
    for (let ch of s1) {
        freq[ch] = (freq[ch] || 0) + 1
    }
    for (let ch of s2) {
        if (!freq[ch]) {
            return false
        } freq[ch]--
    }
    return true
}
console.log(Isanagram("good", "doog"))
console.log(Isanagram("hello", "world"))

File: python/isAnagram.py
def isAnagram(s1, s2):
    if len(s1) != len(s2):
        return False
    freq={}
    for ch in s1:
        if ch in freq:

            freq[ch]+=1
        else:
            freq[ch]=1

    for ch in s2:
        if ch not in freq or freq[ch]==0:
            return False
    return True

print(isAnagram("fuck", "kucf"))
print(isAnagram("hello", "helli"))